packages/api-server/src/methods/modules/prof.ts

```diff
@@ -0,0 +1,34 @@
+import { cpuProf } from "../../decorator";
+import v8Profiler from "v8-profiler-next";
+import fs from "fs";
+import path from "path";
+
+const PROF_TIME_MS = 25000; // 25s
+
+export class Prof {
+  constructor() {}
+
+  @cpuProf(PROF_TIME_MS)
+  async cpu(_args: []): Promise<string> {
+    return "ok";
+  }
+
+  async heap() {
+    const createHeadDumpFile = async (fileName: string) => {
+      return new Promise((resolve, reject) => {
+        const file = fs.createWriteStream(fileName);
+        const snapshot = v8Profiler.takeSnapshot();
+        const transform = snapshot.export();
+        transform.pipe(file);
+        transform.on("finish", () => {
+          snapshot.delete.bind(snapshot);
+          resolve(fileName);
+        });
+        transform.on("error", reject);
+      });
+    };
+    const name = path.join(`${Date.now()}.heapsnapshot`);
+    await createHeadDumpFile(name);
+    return "ok";
```

Comment:
What about returning the file path of the heap snapshot?

Reply:
I am not so sure. assume the RPC module is accidentally open in production, this file path might expose some info ? the file will be located in the root project by the way.

Reply:
I don't think the file path would expose any secret info.
Or returns the filename only?
Explicit file names are helpful when debugging.

Reply:
return filename will be great, updated